rediscontroller/controller.py

```diff
@@ -1,27 +1,75 @@
 import os
 import random
 import subprocess
+from numbers import Integral
 
-from rediscontroller.utils import get_install_prefix, which
+from rediscontroller.utils import get_install_prefix, which, changed_dir
+from rediscontroller.utils import reserve_port
 
 REDIS_PORT = 65535
 
 
+def _careful_start_redis(redis_server_path, data_directory, config_file_path, redis_port):
+    """
+    Tries to start redis at the specified directory, port, with the specified
+    config file. Returns None if there was a port conflict and raises a
```

Comment:
while that's true, this is meant to be an internal function and I wanted to distinguish the case where redis could not be started because of port collisions vs 'any other reason'. Hence the 'any other reason' raises runtime error whereas the port collision returns a None. (So that the calling function can deal with this case separately)

Reply:
ok